src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -3051,6 +3051,36 @@ bool AdaptDispatch::DoFinalTermAction(const std::wstring_view string)
     return false;
 }
 
+bool AdaptDispatch::DoUrxvtAction(const std::wstring_view string)
+{
+    // This is not implemented in conhost.
+    if (_api.IsConsolePty())
+    {
+        // Flush the frame manually, to make sure marks end up on the right line, like the alt buffer sequence.
+        _renderer.TriggerFlush(false);
+        return false;
+    }
+
+    const auto parts = Utils::SplitString(string, L';');
+
+    if (parts.size() < 1)
+    {
+        return false;
+    }
+
+    const auto action = til::at(parts, 0);
+
+    if (action == L"notify")
+    {
+        const std::wstring_view title = parts.size() > 1 ? til::at(parts, 1) : L"";
+        const std::wstring_view body = parts.size() > 2 ? til::at(parts, 2) : L"";
+        _api.SendNotification(title, body);
```

Comment:
This is not a big deal, but ideally the body should include everything after the third semicolon, so you can potentially include a semicolon in the body content. That's the way the original urxvt extension worked, and most implementations seem to support that (although not all). I wouldn't worry if it's not an easy fix though.